pkg/e2e/verify/namespace_webhook.go

```diff
@@ -159,6 +163,80 @@ var _ = ginkgo.Describe(namespaceWebhookTestName, func() {
 				Expect(err).NotTo(HaveOccurred())
 			}
 		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("dedicated admin can not create PrometheusRule", func() {
+			for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, DUMMY_USER, "dedicated-admins", h)
+				err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+					if err != nil {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return false, nil
```

Comment:
I'm feeling like maybe there's room for optimization in these "we expect an error to happen" tests. If I'm reading it right, we expect `Create()` to error in this scenario, and this test continues to try a `Create()` for three minutes and keeps the poll going until it eventually times out after three minutes, returning an error - which is then the criteria for test success.

Aside from that, if the `Create()` call does succeed, it's returning `true, nil`, so afterwards `err` is going to be nil anyway. Which seems not correct when it then goes to check if the error `HasOccurred()`. 

Like, I'm thinking this needs to be flipped on its head and be something like
```
err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
   _, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
   if err != nil {
      log.Printf("failed creating prometheus rules: %v", err)
      return true, nil
   }
   return false, err
})
Expect(err).NotTo(HaveOccurred(), "PrometheusRules could be created")
```

I am not sure what use we get from verifying that `Create()` continues to fail over and over for X minutes.

Reply:
Some more quick thoughts

- We could check and see what error the API will actually return in the scenario where the user doesn't have the RBAC. Only treat the specific RBAC-related error you expect from the API as the "success" criteria. And if the error is _not_ that, then that is cause to go back and have the poller retry.
- And on the topic of the poller, the three minute wait is a bit too generous. This should succeed very quickly. One minute max is probably enough.